pandas/_libs/tslibs/period.pyx

```diff
@@ -1756,9 +1758,40 @@ cdef class _Period(PeriodMixin):
 
     def __cinit__(self, int64_t ordinal, BaseOffset freq):
         self.ordinal = ordinal
-        self.freq = freq
+        self._freq = freq
         self._dtype = PeriodDtypeBase(freq._period_dtype_code, freq.n)
 
+    @property
+    def freq(self) -> BaseOffset:
```

Comment:
this change is going to (slightly) slow down a lot of methods.  im not sure its worth it